homeassistant/components/asuswrt/bridge.py

```diff
@@ -73,20 +72,17 @@ def _handle_errors_and_zip(func: _FuncType) -> _ReturnFuncType:
 
         @functools.wraps(func)
         async def _wrapper(self: _AsusWrtBridgeT) -> dict[str, Any]:
-            try:
-                data = await func(self)
-            except exceptions as exc:
-                raise UpdateFailed(exc) from exc
+            data = await func(self)
 
             if keys is None:
                 if not isinstance(data, dict):
-                    raise UpdateFailed("Received invalid data type")
+                    raise TypeError("Received invalid data type")
```

Comment:
Please define a custom exception that inherits `HomeAssistantError` and use in these cases.

Reply:
I looked around in the integration and these methods seem actually to be used inside the coordinator. It's ok to keep raising `UpdateFailed` here for now. But I would recommend refactoring the integration and define a custom coordinator class instead that raises `UpdateFailed` from inside `_async_update_data`. That would be more appropriate and follow the intended design of the coordinator.

Reply:
Ie this call should instantiate a custom coordinator class.

https://github.com/home-assistant/core/blob/b35d7a348bbcb01d1f202e2840558098fc44797c/homeassistant/components/asuswrt/router.py#L86

Reply:
And then we'd move the error handling to the coordinator too.

Reply:
Ok, for now I restored the `UpdateFailed` exception in decorator. I will leave definition of custom decorator class for a next PR.